src/misc/KeyManager.js

```diff
@@ -183,12 +183,7 @@ class KeyManager {
 	unregisterShortcuts(shortcuts: Shortcut[]) {
 		for (let s of shortcuts) {
 			let id = this._createKeyIdentifier(s.key.code, s.ctrl, s.alt, s.shift, s.meta)
-			this._shortcuts.forEach(s => {
-				if (s.id === id) {
-					let i = this._shortcuts.indexOf(s)
-					this._shortcuts.splice(i, 1)
-				}
-			})
+			findAllAndRemove(this._shortcuts, (s) => s.id === id)
```

Comment:
After looking at this again I think we have a chance to improve the key manager while we're changing it anyway - It seems like we are redundantly tracking shortcuts in both an array (`_shortcuts`) and a map (`_keyToShortcut`) for no reason (and same for modal shortcuts). It would be good to clean this up a bit, and I don't think it would be too extensive of a change

Reply:
Yes, i had noticed this as well. We need to make sure that we reset to the correct shortcuts when we close a modal though.